src/Compilers/CSharp/Portable/Emitter/Model/PEAssemblyBuilder.cs

```diff
@@ -56,9 +62,22 @@ internal abstract class PEAssemblyBuilderBase : PEModuleBuilder, Cci.IAssemblyRe
 
         public override ISourceAssemblySymbolInternal SourceAssemblyOpt => _sourceAssembly;
 
-        internal override ImmutableArray<NamedTypeSymbol> GetAdditionalTopLevelTypes()
+        internal override ImmutableArray<NamedTypeSymbol> GetAdditionalTopLevelTypes(DiagnosticBag diagnostics)
         {
-            return _additionalTypes;
+            var builder = ArrayBuilder<NamedTypeSymbol>.GetInstance();
+            builder.AddRange(_additionalTypes);
+
+            CreateEmbeddedAttributesIfNeeded(diagnostics);
```

Comment:
If there are errors generated embedded attributes, the first time `GetAdditionalTopLevelTypes` is called, those errors will be returned, but subsequent calls will not get those errors. That seems somewhat fragile. Should `CreateEmbeddedAttributesIfNeeded` be called outside of `GetAdditionalTopLevelTypes`, and called at most once per emit?